x-pack/plugins/enterprise_search/public/applications/workplace_search/components/overview/overview_logic.ts

```diff
@@ -0,0 +1,204 @@
+      false,
+      {
+        setServerData: (_: IOverviewValues, { canCreateInvitations }: IOverviewValues) =>
+          canCreateInvitations,
+      },
+    ],
+    flashMessages: [
+      {},
+      {
+        setFlashMessages: (_: IOverviewValues, { flashMessages }: IOverviewValues) => flashMessages,
+      },
+    ],
+    hasUsers: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { hasUsers }: IOverviewValues) => hasUsers,
+      },
+    ],
+    hasOrgSources: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { hasOrgSources }: IOverviewValues) =>
+          hasOrgSources,
+      },
+    ],
+    canCreateContentSources: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { canCreateContentSources }: IOverviewValues) =>
+          canCreateContentSources,
+      },
+    ],
+    isOldAccount: [
+      false,
+      {
+        setServerData: (_: IOverviewServerData, { isOldAccount }: IOverviewValues) => isOldAccount,
+      },
+    ],
+    sourcesCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { sourcesCount }: IOverviewValues) => sourcesCount,
+      },
+    ],
+    pendingInvitationsCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { pendingInvitationsCount }: IOverviewValues) =>
+          pendingInvitationsCount,
+      },
+    ],
+    accountsCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { accountsCount }: IOverviewValues) =>
+          accountsCount,
+      },
+    ],
+    personalSourcesCount: [
+      0,
+      {
+        setServerData: (_: IOverviewServerData, { personalSourcesCount }: IOverviewValues) =>
+          personalSourcesCount,
+      },
+    ],
+    activityFeed: [
+      [],
+      {
+        setServerData: (_: IOverviewServerData, { activityFeed }: IOverviewValues) => activityFeed,
+      },
+    ],
+    dataLoading: [
+      true,
+      {
+        setServerData: () => false,
+        setHasErrorConnecting: () => false,
+      },
+    ],
+    hasErrorConnecting: [
+      false,
+      {
+        setHasErrorConnecting: (_: IOverviewServerData, { hasErrorConnecting }: IOverviewValues) =>
+          hasErrorConnecting,
+      },
+    ],
+  }),
+  selectors: ({ selectors }: { selectors: IOverviewValues }) => ({
+    hideOnboarding: [
+      () => [
+        selectors.hasUsers,
+        selectors.hasOrgSources,
+        selectors.isOldAccount,
+        selectors.organization,
+      ],
+      (hasUsers: boolean, hasOrgSources: boolean, isOldAccount: boolean, org: IOrganization) =>
+        hasUsers && hasOrgSources && org.name !== org.defaultOrgName && isOldAccount,
+    ],
+    statsColumns: [
+      () => [selectors.isFederatedAuth],
+      (isFederatedAuth: boolean) => (isFederatedAuth ? 'halves' : 'fourths'),
+    ],
+  }),
+  listeners: ({ actions }: IListenerParams) => ({
+    initializeOverview: async ({ http }: { http: HttpSetup }) => {
```

Comment:
After poking at this a bit I think I'd like to continue just passing in `http` for now in this PR if that's cool 🤔 After looking some more at Kea's `defaults` options, I'm not totally sure that will work the way we expect - so we'd have to instantiate a new kea() store just for storing `http`, and connect / import all logic files to that, and at that point we might as well just convert the entire KibanaContext to Kea from React.Context 🤷‍♀️

I'll continue poking at it but so far from the amount of changes I'm having to do, this will almost certainly need a separate PR